### PR TITLE
Adds support for non-scalar output types

### PR DESCRIPTION
**Pull request by efernandez**
And also adds `--import` arg.

### CODE REVIEW

**Comment by po1** [comment]
I fear this is going to fail with a string. Can you test it?
If it does fail, maybe you should add a check for `not isinstance(res, basestring)`

**Comment by dirk-thomas** [comment]
Please rebase to head in order to make the PR mergable.

**Comment by efernandez** [comment]
@po1 I changed the check to use `isinstance(res, (list, tuple))`.
@dirk-thomas rebased on top of `indigo-devel`.

File: tools/topic_tools/scripts/transform [hunk 1]
```diff
@@ -51,9 +50,22 @@ class TopicOp:
...
+        )
 
         args = parser.parse_args()
 
+        self.modules = {}
+        for module in args.modules:
+            try:
+                mod = importlib.import_module(module)
+            except ImportError:
+                print('Failed to import module:', module)
```
Comment (dirk-thomas) [inline tools/topic_tools/scripts/transform:65]: Please print error messages to stderr, like this:

```
print('...', file=sys.stderr)
```

File: tools/topic_tools/scripts/transform [hunk 2]
```diff
@@ -61,17 +73,17 @@ class TopicOp:
...
 
     def callback(self, m):
         if self.input_fn is not None:
             m = self.input_fn(m)
 
         try:
-            res = eval(self.expression, {}, {'m': m})
+            res = eval("{}".format(self.expression), self.modules, {'m': m})
         except NameError as e:
             print("Expression using variables other than 'm':", e.message)
```
Comment (dirk-thomas) [inline tools/topic_tools/scripts/transform:88]: Same here.

**Comment by dirk-thomas** [comment]
Wouldn't it be more flexible to allow the user to use `import` within its expression? Then you wouldn't need the import option at all.

**Comment by dirk-thomas** [comment]
Since we are releasing ros_comm _now_ I will merge this PR with my print-related comments: b8e7b987b0621c7a126c176027a768232da28ac7

I would still suggest considering my other comment regarding the imports. A potential change can be merged for the next patch release then.

**Comment by efernandez** [comment]
@dirk-thomas  Regarding the imports, since I use `eval` to evaluate the given `expression`, which must be a Python expression and not a statement (as `exec` will support). For that reason, we cannot use `import module` inside `eval`, and that's why I added the `--import` argument. IMHO, using `exec` will require an output variable to assign the given expression to, so the user must do that.

These are the current examples using `eval`:

<pre>
$ rosrun topic_tools transform /imu/orientation/x /x_str std_msgs/String 'str(m)'
$ rosrun topic_tools transform /imu/orientation/x /x_in_degrees std_msgs/Float64 -- '-numpy.rad2deg(m)'
$ rosrun topic_tools transform /imu/orientation /norm std_msgs/Float64 'numpy.sqrt(numpy.sum(numpy.array([m.x, m.y, m.z, m.w])))'
$ rosrun topic_tools transform /imu/orientation /norm std_msgs/Float64 'numpy.linalg.norm([m.x, m.y, m.z, m.w])'
$ rosrun topic_tools transform /imu/orientation /euler geometry_msgs/Vector3 'tf.transformations.euler_from_quaternion([m.x, m.y, m.z, m.w])' --import tf
</pre>

Using `exec` they should be something like (using `out` and `in`, instead of simply `m`):

<pre>
$ rosrun topic_tools transform /imu/orientation/x /x_str std_msgs/String 'out = str(in)'
$ rosrun topic_tools transform /imu/orientation/x /x_in_degrees std_msgs/Float64 -- 'import numpy; out = -numpy.rad2deg(in)'
$ rosrun topic_tools transform /imu/orientation /norm std_msgs/Float64 'from numpy import *; out = sqrt(sum(array([in.x, in.y, in.z, in.w]))))'
$ rosrun topic_tools transform /imu/orientation /norm std_msgs/Float64 'from numpy.linalg import norm; out = norm([in.x, in.y, in.z, in.w])'
$ rosrun topic_tools transform /imu/orientation /euler geometry_msgs/Vector3 'from tf.transformations import euler_from_quaternion; out = euler_from_quaternion([in.x, in.y, in.z, in.w])'
</pre>

Basically, we need `out =`, but an advantage is that we can do `from module import *`.

@dirk-thomas @po1 What do you think? It might actually be better to use `exec`, since it's more clear and flexible for the user: just write a Python statement (or many).

**Comment by po1** [comment]
We have to consider the point where it will simply be easier for the user to fire ipython and to type a few lines in an interactive interpreter rather than to fiddle with the syntax of this tool.

I like what the tool does for simple cases, but the syntax can already be cumbersome. My opinion is that it better stay simple.

**Comment by dirk-thomas** [comment]
I am okay keeping the imports as-is. I was just throwing in the idea. Bit the rational to keep it simpler is a good point.

**Comment by efernandez** [comment]
Alright.
BTW, could this tool be back-ported to hydro-devel?

**Comment by dirk-thomas** [comment]
Since it is newly introduced I will consider it for back porting for the next Hydro patch release.

Can you please add documentation about it to the wiki so that users know about the tool?

**Comment by efernandez** [comment]
@dirk-thomas wiki doc added to:
http://wiki.ros.org/topic_tools
http://wiki.ros.org/topic_tools/transform

**Comment by dirk-thomas** [comment]
Great - thank you!
